sources/iTermURLActionFactory.m

```diff
@@ -309,6 +309,9 @@ - (void)trySecureCopy {
 #pragma mark - Sub-factories
 
 - (URLAction *)urlActionForPrompt {
+    if (![iTermAdvancedSettingsModel enableCmdClickPromptForShowCommandInfo]) {
+        return nil;
+    }
```

Comment:
For my use case this felt cleaner than reordering the phases in the factory (so that Smart Selection is found first), but I'm open to the other path also.